### DISCUSSION TITLE
DISTO simulation in dB

### DISCUSSION THREAD

**Discussion by bnilsson11** [ra3xdh/qucs_s#964] [Q&A] [answered]
I would like to display the distorsion components h2 and h3 in dB relative to the fundamental.
I can display ac1, h2 and h3 as a function of frequency, following the ngspice manual and some examples that I have been given.
However, I cannot find a way to do e.g. h3/ac1 to get a relative value, preferably in dB relative the fundamental.
Help would be greatly appreciated.

**Comment by ra3xdh** [comment]
Have you tried to define the equation and specify the simulation referencing the distortion simulation? See the variable `h3DB` in the attached screenshot.
![image](https://github.com/user-attachments/assets/5db7d885-35b9-4d44-9d52-0a67d30d72ef)

**Comment by bnilsson11** [comment]
I probably misunderstand something here.
In order to interpret the level of the harmonics properly, they need to be expressed  in relation to the fundamental. (E.g. traditionally in % of the total output, similar to how the total THD % is expressed).

The amplifier "gain", for example, should be calculated as dB(out/in). 

h2.out and h3.out are absolute values in volts, right? 
To pass h2.out and h3.out thru the dB() function would not change anything.

To get the ratio of distortion components to the output level, you need to do something similar to "h2reldb=db(h2.out/h1.out)".
The ngspice manual touches this in the DISTO section, but not detailed enough to be helpful to me:

> 11.3.3
> To obtain HD2 and HD3, one must divide by the corresponding AC values
> at F 1, obtained from an .ac line.

I do not know how to access the fundamental "out" in the DISTO1 simulation context, and "out" from AC1 is not available in DISTO1.

**Comment by ivandi69** [comment, marked as answer]
You can use a Nutmeg script:
![image](https://github.com/user-attachments/assets/2fb59676-808c-4628-be53-f9542cf8fbe3)
[Distortion2.sch.txt](https://github.com/user-attachments/files/17549714/Distortion2.sch.txt)

**Comment by bnilsson11** [comment]
Thanks, that solved it for me.
I need to take a course in nutmeg scripting.
